lib/formulator.ex

```diff
@@ -14,25 +14,58 @@ defmodule Formulator do
     When given `false`, does not create a label tag. Instead, an `aria-label`
     attribute is added to the input to improve accessibility.
 
+    * `:validate` - Defaults to application config. When provided a form created with an
+    ecto changeset that contains validations, then Formulator will add HTML5
+    validation attributes (except regex).
+
+    * `:validate_regex` - Defaults to application config. Like option `:validate`, except
+    this will add a pattern HTML5 validation. This should work with most simple
+    regex patterns, but the browser's regex engine may differ from Erlang's.
+
   ## Examples
 
   Basic input:
-      <%= input form, :email %>
-      #=> <label for="user_email">Email</label>
-      #=> <input id="user_email" name="user[email]" type="text" value="">
+      <%= input form, :name %>
+      #=> <label for="user_name">Name</label>
+      #=> <input id="user_name" name="user[name]" type="text" value="">
 
   Without a label:
-      <%= input form, :email, label: false %>
-      #=> <input id="user_name" name="user[email]" aria-label="email" type="text" value="">
+      <%= input form, :name, label: false %>
+      #=> <input id="user_name" name="user[name]" aria-label="name" type="text" value="">
 
   Passing other options:
-      <%= input form, :email, label: [class: "control-label"] %>
-      #=> <label class="control-label" for="user_email">Email</label>
-      #=> <input id="user_email" name="user[email]" type="text" value="">
+      <%= input form, :name, label: [class: "control-label"] %>
+      #=> <label class="control-label" for="user_name">Name</label>
+      #=> <input id="user_name" type="text" name="user[name]" value="">
+
+  Using different input types:
+      <%= input form, :email_address, as: :email, placeholder: "your@email.com", class: "my-email-class", label: [class: "my-email-label-class"] %>
+      #=> <label class="my-email-label-class" for="user_email_address">Email Address</label>
+      #=> <input id="user_email_address" type="email" name="user[email_address]" placeholder: "your@email.com" value="" class="my-email-class">
+
+  Or a number input:
+      <%= input form, :count, as: :number %>
```

Comment:
Should we automatically infer this with:`Phoenix.HTML.Form.input_type/2` ?

Reply:
Also, if you agree, I'm totally fine pushing that to another PR

Reply:
It's not related to this PR, so ya, it should go into a different one. I think inferring it makes sense, but I think that'd be an odd behavior if we only did it for numbers. I assume we would infer from an Ecto.Schema, and if we do it for numbers, we should probably do it for booleans, and maybe on emails, and passwords, etc.